codec-http2/src/main/java/io/netty/handler/codec/http2/HpackEncoder.java

```diff
@@ -257,15 +257,9 @@ private void encodeStringLiteral(ByteBuf out, CharSequence string) {
             hpackHuffmanEncoder.encode(out, string);
         } else {
             encodeInteger(out, 0x00, 7, string.length());
-            if (string instanceof AsciiString) {
-                // Fast-path
-                AsciiString asciiString = (AsciiString) string;
-                out.writeBytes(asciiString.array(), asciiString.arrayOffset(), asciiString.length());
-            } else {
-                // Only ASCII is allowed in http2 headers, so its fine to use this.
-                // https://tools.ietf.org/html/rfc7540#section-8.1.2
-                out.writeCharSequence(string, CharsetUtil.ISO_8859_1);
-            }
+            // Only ASCII is allowed in http2 headers, so its fine to use this.
+            // https://tools.ietf.org/html/rfc7540#section-8.1.2
+            ByteBufUtil.writeAscii(out, string);
```

Comment:
Why not just leave it as it is ? To remove code-duplication ? Also This not belongs to this PR imho